pybamm/solvers/dae_solver.py

```diff
@@ -125,6 +97,78 @@ def jacobian(t, y):
 
         pybamm.logger.info("Finish solving {}".format(model.name))
 
+    def set_up(self, model):
+        """Unpack model, perform checks, simplify and calculate jacobian.
+
+        Parameters
+        ----------
+        model : :class:`pybamm.BaseModel`
+            The model whose solution to calculate. Must have attributes rhs and
+            initial_conditions
+
+        Returns
+        -------
+        concatenated_rhs : :class:`pybamm.Concatenation`
+            Right-hand side of differential equations
+        concatenated_algebraic : :class:`pybamm.Concatenation`
+            Algebraic equations, which should evaluate to zero
+        y0 : :class:`numpy.array`
+            Vector of initial conditions
+        events : list of :class:`pybamm.Symbol`
+            List of events at which the model should terminate
+        jac : :class:`pybamm.SparseStack`
+            Jacobian matrix for the differential and algebraic equations
+
+        Raises
+        ------
+        :class:`pybamm.SolverError`
+            If the model contains any algebraic equations (in which case a DAE solver
+            should be used instead)
+        """
+        if len(model.algebraic) == 0:
+            raise pybamm.SolverError(
```

Comment:
Should this be an error? It is valid to solve an ODE system with a DAE solver (not sure if it is actually useful though!)

Reply:
Yeah I just got lazy and said this should be an error ... do you think it should give a warning or just pass?

Reply:
Worth noting that sundials cvode doesn't allow for ODEs with a mass matrix (i.e. of the form `M*y_dot = f(t,y)`), it requires that you invert to give `y_dot` explicitly. You can pass ODEs like `M*y_dot = f(t,y)` to ida, so I guess people might want to do this if they don't like the scipy solver for some reason...